source/libraries/CompileSolidity.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as solc from 'solc';
+
+class SolidityContractCompiler {
+    private compiler: any = solc;
+    private contractInputDirectoryPath: string;
+    private contractOutputDirectoryPath: string;
+    private contractOutputFileName: string;
+
+    public constructor(contractInputDirectoryPath: string, contractOutputDirectoryPath: string, contractOutputFileName: string) {
+        this.contractInputDirectoryPath = contractInputDirectoryPath;
+        this.contractOutputDirectoryPath = contractOutputDirectoryPath;
+        this.contractOutputFileName = contractOutputFileName;
+    }
+
+    public compileContracts() {
+        // Compile all contracts in this.inputDirectoryPath
+        let inputJSON: any = this.generateCompilerInput();
+        let compilerOutput: any = this.compiler.compile(inputJSON);
+        if (compilerOutput.errors.length > 0) {
+            console.log("The compiler encountered the following warnings/errors:");
+            for (let error of compilerOutput.errors) {
+                console.log(error);
+            }
+            // Do not return automatically in case there are only warnings and no errors
+        }
+
+        // Create output directory (if it doesn't exist) and save contract bytecodes to single file
+        this.createOutputDirectory();
+        let contractOutputFilePath: string = this.contractOutputDirectoryPath + "/" + this.contractOutputFileName;
+        let wstream: any = fs.createWriteStream(contractOutputFilePath);
+        for (let contract in compilerOutput.contracts) {
+            wstream.write(compilerOutput.contracts[contract].bytecode);
+        }
+
+        console.log("Saved " + contractOutputFilePath);
+    }
+
+    private getContractFilePaths(directoryPath){
+        let contractFilePaths: string[] = [];
+        if (!fs.existsSync(directoryPath)) {
+            console.log("Invalid directory path: ", directoryPath);
+        } else {
+            let filesInCurrentDirectory: any = fs.readdirSync(directoryPath);
+            for (let file of filesInCurrentDirectory) {
+                let filePath: string = path.join(directoryPath, file);
+                let stat: any = fs.lstatSync(filePath);
+                if (stat.isDirectory()) {
+                    let filePaths: string[] = this.getContractFilePaths(filePath);
+                    for (let filePath of filePaths) {
+                        contractFilePaths.push(filePath);
+                    }
+                } else if (stat.isFile() && filePath.split(".").pop() == "sol") {
+                    contractFilePaths.push(filePath);
+                }
+            }
+        }
+
+        return contractFilePaths;
+    }
+
+    private generateCompilerInput() {
+        let inputJSON: any = {
```

Comment:
We should create a typing file for solc that defines the structure of this so we get type checking, static analysis and auto-complete on this object in a TypeScript aware editor and compiler errors the rest of the time.  Writing type definition files is a bit annoying because you have to figure out the right kind of definition file to author based on the library, but once you have the structure down writing the definition is pretty easy.  Let me know when you want to work on this and I can help out.

Reply:
Ok, I can work on this today.  I suppose we might as well do this before moving on to creating a unit test?

Reply:
Same comment pretty much anywhere `any` is used for a type.  It takes a bit of up-front work to add custom type definitions, but in the long run I believe it is well worth it.

Reply:
Yeah, writing tests will be easier when we have strong types.